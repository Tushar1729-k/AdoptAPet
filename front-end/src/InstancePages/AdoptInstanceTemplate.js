import React from 'react'
import { Container, Row, Col, Card, Badge, ListGroup, Button } from 'react-bootstrap'
import { Link, BrowserRouter as Router, Switch, useHistory } from "react-router-dom";
import CustomMap from '../GeneralPages/GoogleMap';
import PropTypes from 'prop-types'
import { InfoCircleFill, PatchPlusFill } from 'react-bootstrap-icons';

const AdoptInstanceTemplate = ({attributes, fetchPage}) => {
    const services = attributes.services.split(",")
    const badgeTypes = ["secondary", "success", "danger", "warning", "info", "light", "dark"]
    let history = useHistory();

    const redirect = (type, num, path) => {
        fetchPage(type, num)
        history.push(path)
    }
    return (
        <div>
            <div style={{paddingLeft: "15vw", paddingRight: "15vw", paddingTop: "4vh"}}>
            <Card style={{ width: '70vw' }}>
            <Card.Body>
                <Card.Title style={{fontSize: '6vh'}}>{attributes.name}</Card.Title>
                <Card.Subtitle style={{fontSize: '4vh'}} className="mb-2 text-muted">{attributes.type}</Card.Subtitle>
                <Card.Text>
                    <Container>
                        <Row>
                            <ListGroup variant="flush">
                                <ListGroup.Item>Species Available : {attributes.species}</ListGroup.Item>
                                <ListGroup.Item>
                                    Services Available :
                                    <div>
                                        <InfoCircleFill/>{' '}
                                        <Badge pill bg="primary">
                                            <PatchPlusFill/>{' '}Adoption
                                        </Badge>
                                        {Array.from({length: services.length}).map((_, idx) => (
                                            <Badge key={idx} pill bg={badgeTypes[idx]}>
                                                <PatchPlusFill/>{' '}{services[idx]}
                                            </Badge>
                                        ))}
                                    </div>
                                    {attributes.services}
                                </ListGroup.Item>
                                <ListGroup.Item>Address : {attributes.address}</ListGroup.Item>
                                <ListGroup.Item>City : {attributes.city}</ListGroup.Item>
                                <ListGroup.Item>State : {attributes.state}</ListGroup.Item>
                                <ListGroup.Item>Zip Code : {attributes.zip}</ListGroup.Item>
                                <ListGroup.Item>Phone : {attributes.phone}</ListGroup.Item>
                                <ListGroup.Item>Email : {attributes.email}</ListGroup.Item>
                                <ListGroup.Item>Website : {attributes.site}</ListGroup.Item>
                            </ListGroup>
                        </Row>
                    </Container>
                </Card.Text>
                <Row>
                    <div style={{width: '100vw', height: '100vh'}}>
                    <CustomMap googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=
                                              AIzaSyCD6X1ARbFEh9eXKDW4EbN-kNIHz-_dlaM`}
                        lat={attributes.lat}
                        lng={attributes.lon}
                        loadingElement={<div style={{height: "100%"}}></div>}
                        containerElement={<div style={{height: "100%"}}></div>}
                        mapElement={<div style={{height: "100%"}}></div>}
                    />
                    </div>
                </Row>
                <Row style={{paddingTop: '2vh'}}>
                    <Col>
                    <h5>See breeds this center carries</h5>
                    {attributes.speciesBreeds.map((sb, idx) => (
                        <Button key={idx} variant="link" style={{textDecoration: 'none'}} onClick={() => redirect("sb", sb.api_id, `/sbmodel/${sb.api_id}`)}>
                            <h5>{sb.breed_name}</h5>
                        </Button>
                    ))}
                    </Col>
                    <Col>
                    <h5>Pets available at this center
                    </h5>
                        {attributes.pets.map((pet, idx) => (
                            <Button key={idx} variant="link" style={{textDecoration: 'none'}} onClick={() => redirect("ap", pet.api_id, `/apmodel/${pet.api_id}`)}>
                                <h5>Take a look at {pet.name}</h5>
                            </Button>
                        ))}
                    </Col>
                </Row>
            </Card.Body>
            </Card>
            </div>
        </div>
    )
}

// Set defaults of props here.
AdoptInstanceTemplate.defaultProps = {
    attributes: { name: '', address: '', city: '', state: '', zip: '',
                  phone: '', email: '', site: 'NA', species: '',
                  services: '', petId: '', sbId: '', mapSrc: '', imgSrc: '',
                  speciesBreeds: [{api_id: 1, name: ''}], pets: [{api_id: 1, name: ''}],
                  lat: 0, lon: 0}
}
// Set type of the prop here.
AdoptInstanceTemplate.propTypes = {
	attributes: PropTypes.object,
    fetchPage: PropTypes.func
}

export default AdoptInstanceTemplate
