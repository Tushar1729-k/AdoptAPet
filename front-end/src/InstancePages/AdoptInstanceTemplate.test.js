import React from 'react';
import AdoptInstanceTemplate from './AdoptInstanceTemplate';
import renderer from 'react-test-renderer';
import terrier1 from '../Images/pitbull_terrier_1.jpeg'

test("Adoption Center Instance Page Properly Rendered", () => {
    const component = renderer.create(
        <AdoptInstanceTemplate
            attributes={{ name: "Pets for Life", address: "187 Star War Way",
            city: "Austin", state: "TX", 
            zip: "78705", phone: "123-456-7890", 
            email: "pets4life@gmail.com",
            site: "www.pets4life.com", species: "Dogs, Cats",
            services: "grooming, training", speciesBreeds: [{api_id: 12}],
            pets: [{api_id: 1, name: 'Jack'}, {api_id: 2, name: 'John'}],
            imgSrc: ""
        }}
        />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})

test("Adoption Center Instance Page Properly Rendered With Image", () => {
    const component = renderer.create(
        <AdoptInstanceTemplate
            attributes={{ name: "Pets for Life", address: "187 Star War Way",
            city: "Austin", state: "TX", 
            zip: "78705", phone: "123-456-7890", 
            email: "pets4life@gmail.com",
            site: "www.pets4life.com", species: "Dogs, Cats",
            services: "grooming, training", speciesBreeds: [{api_id: 12}],
            pets: [{api_id: 1, name: 'Jack'}, {api_id: 2, name: 'John'}],
            imgSrc: terrier1
        }}
        />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})