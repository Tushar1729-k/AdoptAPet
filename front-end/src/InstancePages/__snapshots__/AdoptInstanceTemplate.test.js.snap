// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adoption Center Instance Page Properly Rendered 1`] = `
<div>
  <div
    style={
      Object {
        "paddingLeft": "15vw",
        "paddingRight": "15vw",
        "paddingTop": "4vh",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#00008b",
          "color": "white",
          "width": "70vw",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
          style={
            Object {
              "fontSize": "6vh",
            }
          }
        >
          Pets for Life
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
          style={
            Object {
              "fontSize": "4vh",
            }
          }
        />
        <p
          className="card-text"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="list-group list-group-flush"
                onKeyDown={[Function]}
              >
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Services Available :
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412-1 4.705c-.07.34.029.533.304.533.194 0 .487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703 0-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381 2.29-.287zM8 5.5a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
                      />
                    </svg>
                     
                    <span
                      className="badge rounded-pill bg-primary"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.067.87a2.89 2.89 0 0 0-4.134 0l-.622.638-.89-.011a2.89 2.89 0 0 0-2.924 2.924l.01.89-.636.622a2.89 2.89 0 0 0 0 4.134l.637.622-.011.89a2.89 2.89 0 0 0 2.924 2.924l.89-.01.622.636a2.89 2.89 0 0 0 4.134 0l.622-.637.89.011a2.89 2.89 0 0 0 2.924-2.924l-.01-.89.636-.622a2.89 2.89 0 0 0 0-4.134l-.637-.622.011-.89a2.89 2.89 0 0 0-2.924-2.924l-.89.01-.622-.636zM8.5 6v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 1 0z"
                        />
                      </svg>
                       
                      Adoption
                    </span>
                    <span
                      className="badge rounded-pill bg-secondary"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.067.87a2.89 2.89 0 0 0-4.134 0l-.622.638-.89-.011a2.89 2.89 0 0 0-2.924 2.924l.01.89-.636.622a2.89 2.89 0 0 0 0 4.134l.637.622-.011.89a2.89 2.89 0 0 0 2.924 2.924l.89-.01.622.636a2.89 2.89 0 0 0 4.134 0l.622-.637.89.011a2.89 2.89 0 0 0 2.924-2.924l-.01-.89.636-.622a2.89 2.89 0 0 0 0-4.134l-.637-.622.011-.89a2.89 2.89 0 0 0-2.924-2.924l-.89.01-.622-.636zM8.5 6v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 1 0z"
                        />
                      </svg>
                       
                      grooming
                    </span>
                    <span
                      className="badge rounded-pill bg-success"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.067.87a2.89 2.89 0 0 0-4.134 0l-.622.638-.89-.011a2.89 2.89 0 0 0-2.924 2.924l.01.89-.636.622a2.89 2.89 0 0 0 0 4.134l.637.622-.011.89a2.89 2.89 0 0 0 2.924 2.924l.89-.01.622.636a2.89 2.89 0 0 0 4.134 0l.622-.637.89.011a2.89 2.89 0 0 0 2.924-2.924l-.01-.89.636-.622a2.89 2.89 0 0 0 0-4.134l-.637-.622.011-.89a2.89 2.89 0 0 0-2.924-2.924l-.89.01-.622-.636zM8.5 6v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 1 0z"
                        />
                      </svg>
                       
                       training
                    </span>
                  </div>
                  grooming, training
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Address : 
                  187 Star War Way
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  City : 
                  Austin
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  State : 
                  TX
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Zip Code : 
                  78705
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Phone : 
                  123-456-7890
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Email : 
                  pets4life@gmail.com
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Website : 
                  www.pets4life.com
                </div>
              </div>
            </div>
          </div>
        </p>
        <div
          className="row"
        >
          <div
            style={
              Object {
                "height": "100vh",
                "width": "100vw",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "paddingTop": "2vh",
            }
          }
        >
          <div
            className="col"
          >
            <h5>
              See breeds this center carries
            </h5>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5 />
            </button>
          </div>
          <div
            className="col"
          >
            <h5>
              Pets available at this center
            </h5>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                Take a look at 
                Jack
              </h5>
            </button>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                Take a look at 
                John
              </h5>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Adoption Center Instance Page Properly Rendered With Image 1`] = `
<div>
  <div
    style={
      Object {
        "paddingLeft": "15vw",
        "paddingRight": "15vw",
        "paddingTop": "4vh",
      }
    }
  >
    <div
      className="card"
      style={
        Object {
          "backgroundColor": "#00008b",
          "color": "white",
          "width": "70vw",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
          style={
            Object {
              "fontSize": "6vh",
            }
          }
        >
          Pets for Life
        </div>
        <div
          className="mb-2 text-muted card-subtitle h6"
          style={
            Object {
              "fontSize": "4vh",
            }
          }
        />
        <p
          className="card-text"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="list-group list-group-flush"
                onKeyDown={[Function]}
              >
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Services Available :
                  <div>
                    <svg
                      fill="currentColor"
                      height="1em"
                      viewBox="0 0 16 16"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16zm.93-9.412-1 4.705c-.07.34.029.533.304.533.194 0 .487-.07.686-.246l-.088.416c-.287.346-.92.598-1.465.598-.703 0-1.002-.422-.808-1.319l.738-3.468c.064-.293.006-.399-.287-.47l-.451-.081.082-.381 2.29-.287zM8 5.5a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
                      />
                    </svg>
                     
                    <span
                      className="badge rounded-pill bg-primary"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.067.87a2.89 2.89 0 0 0-4.134 0l-.622.638-.89-.011a2.89 2.89 0 0 0-2.924 2.924l.01.89-.636.622a2.89 2.89 0 0 0 0 4.134l.637.622-.011.89a2.89 2.89 0 0 0 2.924 2.924l.89-.01.622.636a2.89 2.89 0 0 0 4.134 0l.622-.637.89.011a2.89 2.89 0 0 0 2.924-2.924l-.01-.89.636-.622a2.89 2.89 0 0 0 0-4.134l-.637-.622.011-.89a2.89 2.89 0 0 0-2.924-2.924l-.89.01-.622-.636zM8.5 6v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 1 0z"
                        />
                      </svg>
                       
                      Adoption
                    </span>
                    <span
                      className="badge rounded-pill bg-secondary"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.067.87a2.89 2.89 0 0 0-4.134 0l-.622.638-.89-.011a2.89 2.89 0 0 0-2.924 2.924l.01.89-.636.622a2.89 2.89 0 0 0 0 4.134l.637.622-.011.89a2.89 2.89 0 0 0 2.924 2.924l.89-.01.622.636a2.89 2.89 0 0 0 4.134 0l.622-.637.89.011a2.89 2.89 0 0 0 2.924-2.924l-.01-.89.636-.622a2.89 2.89 0 0 0 0-4.134l-.637-.622.011-.89a2.89 2.89 0 0 0-2.924-2.924l-.89.01-.622-.636zM8.5 6v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 1 0z"
                        />
                      </svg>
                       
                      grooming
                    </span>
                    <span
                      className="badge rounded-pill bg-success"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 16 16"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M10.067.87a2.89 2.89 0 0 0-4.134 0l-.622.638-.89-.011a2.89 2.89 0 0 0-2.924 2.924l.01.89-.636.622a2.89 2.89 0 0 0 0 4.134l.637.622-.011.89a2.89 2.89 0 0 0 2.924 2.924l.89-.01.622.636a2.89 2.89 0 0 0 4.134 0l.622-.637.89.011a2.89 2.89 0 0 0 2.924-2.924l-.01-.89.636-.622a2.89 2.89 0 0 0 0-4.134l-.637-.622.011-.89a2.89 2.89 0 0 0-2.924-2.924l-.89.01-.622-.636zM8.5 6v1.5H10a.5.5 0 0 1 0 1H8.5V10a.5.5 0 0 1-1 0V8.5H6a.5.5 0 0 1 0-1h1.5V6a.5.5 0 0 1 1 0z"
                        />
                      </svg>
                       
                       training
                    </span>
                  </div>
                  grooming, training
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Address : 
                  187 Star War Way
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  City : 
                  Austin
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  State : 
                  TX
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Zip Code : 
                  78705
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Phone : 
                  123-456-7890
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Email : 
                  pets4life@gmail.com
                </div>
                <div
                  className="list-group-item"
                  data-rr-ui-event-key={null}
                  onClick={[Function]}
                >
                  Website : 
                  www.pets4life.com
                </div>
              </div>
            </div>
          </div>
        </p>
        <div
          className="row"
        >
          <div
            style={
              Object {
                "height": "100vh",
                "width": "100vw",
              }
            }
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "paddingTop": "2vh",
            }
          }
        >
          <div
            className="col"
          >
            <h5>
              See breeds this center carries
            </h5>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5 />
            </button>
          </div>
          <div
            className="col"
          >
            <h5>
              Pets available at this center
            </h5>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                Take a look at 
                Jack
              </h5>
            </button>
            <button
              className="btn btn-link"
              disabled={false}
              onClick={[Function]}
              style={
                Object {
                  "color": "#add8e6",
                  "textDecoration": "none",
                }
              }
              type="button"
            >
              <h5>
                Take a look at 
                John
              </h5>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
